### PR TITLE
Show warning when git is not installed

### PR DESCRIPTION
**Pull request by yufeih**
#2863

### CODE REVIEW

File: src/docfx/lib/git/GitUtility.cs [hunk 1]
```diff
@@ -128,9 +129,14 @@ private static Task<string> ExecuteQuery(string cwd, string commandLineArgs, Tim
...
             Debug.Assert(!string.IsNullOrEmpty(cwd));
 
-            // todo: check git exist or not
-            var response = await ProcessUtility.Execute("git", commandLineArgs, cwd, timeout, outputHandler);
-            return parser(response);
+            try
+            {
+                return parser(await ProcessUtility.Execute("git", commandLineArgs, cwd, timeout, outputHandler));
+            }
+            catch (Win32Exception ex) when (ProcessUtility.IsNotFound(ex))
```
Comment (herohua) [inline src/docfx/lib/git/GitUtility.cs]: >Win32Exception [](start = 19, length = 14)

this is not cross platform?

Comment (yufeih) [reply src/docfx/lib/git/GitUtility.cs]: this **is** xplat, linux test passed. 

.Net core cannot make breaking changes like this (renaming exception), so it'll still throw Win32Exception on linux and mac.

File: src/docfx/Errors.cs [hunk 2]
```diff
@@ -96,6 +96,9 @@ public static Error RedirectionDocumentIdConflict(IEnumerable<Document> redirect
         public static Error GitShadowClone(string repoPath)
             => new Error(ErrorLevel.Error, "git-shadow-clone", $"Does not support git shallow clone: '{repoPath}'");
 
+        public static Error GitNotFound()
+            => new Error(ErrorLevel.Error, "git-not-found", $"Cannot find git, install git https://git-scm.com/");
```
Comment (superyyrrzz) [inline src/docfx/Errors.cs]: >Error [](start = 36, length = 5)

Warning? Some users may not need git features.

Comment (yufeih) [reply src/docfx/Errors.cs]: when this is executed, it sure is invoking git exe and the user is using git features that depend on git exe

File: src/docfx/lib/git/GitUtility.cs (hunk 1 above)
Comment (herohua) [reply src/docfx/lib/git/GitUtility.cs]: looks good!

---
In reply to: [200887867](https://github.com/dotnet/docfx/pull/3022#discussion_r200887867) [](ancestors = 200887867)

File: src/docfx/lib/git/GitUtility.cs [hunk 3]
```diff
@@ -128,9 +129,14 @@ private static Task<string> ExecuteQuery(string cwd, string commandLineArgs, Tim
...
+            {
+                throw Errors.GitNotFound().ToException();
```
Comment (herohua) [inline src/docfx/lib/git/GitUtility.cs]: >ToException [](start = 43, length = 11)

put ex as inner exception?

**Pull request merged**